influxdb3_server/src/http.rs

```diff
@@ -404,6 +405,18 @@ impl IntoResponse for Error {
                     .body(body)
                     .unwrap()
             }
+            Self::WriteBuffer(err @ WriteBufferError::DatabaseDeleted(_)) => {
+                let err: ErrorMessage<()> = ErrorMessage {
+                    error: err.to_string(),
+                    data: None,
+                };
+                let serialized = serde_json::to_string(&err).unwrap();
+                let body = bytes_to_response_body(serialized);
+                ResponseBuilder::new()
+                    .status(StatusCode::BAD_REQUEST)
```

Comment:
400 bad request is better than 500 internal server error imo. I also debated 403: forbidden but i think 400 is better.